### ISSUE TITLE
Use `Instant` over `SystemTime` to record timings

### ISSUE THREAD

**Issue by rtfeldman** [roc-lang/roc#3575]
[`Instant`](https://doc.rust-lang.org/std/time/struct.Instant.html) is better than `SystemTime` for this, because:
* It's guaranteed to be monotonically increasing (unlike the system clock, which can go backwards if the user changes the OS clock while the program is running)
* Because of this guarantee, `.elapsed()` doesn't return an `Option` that we have to unwrap, unlike `SystemTime::elapsed`

Basically this issue is to use `Instant` everywhere we're currently using `SystemTime`, and then replace `.elapsed().unwrap()` with `.elapsed()`.
